pkg/cloud/azure/assets/cloud-controller-manager-deployment.yaml

```
@@ -53,6 +53,53 @@ spec:
         - effect: NoSchedule
           key: node.kubernetes.io/not-ready
           operator: Exists
+      initContainers:
+        # Merge /etc/kubernetes/cloud.conf (on the host) with secret "azure-cloud-credentials" into "merged-cloud-config" emptydir.
+        - name: azure-inject-credentials
+          image: {{ .images.CloudControllerManagerOperator }}
+          command:
+            - /bin/bash
+            - -c
+            - |
+              #!/bin/bash
+              if [[ -f /etc/cloud-config/apiserver-url.env ]]; then
+                cp /etc/cloud-config/apiserver-url.env /etc/merged-cloud-config/
+              fi
+              exec /azure-config-credentials-injector \
+                --cloud-config-file-path=/etc/cloud-config/cloud.conf \
+                --output-file-path=/etc/merged-cloud-config/cloud.conf \
+                --disable-identity-extension-auth \
+                --enable-azure-workload-identity=true
+          env:
+            - name: AZURE_CLIENT_ID
+              valueFrom:
+                secretKeyRef:
+                  name: azure-cloud-credentials
+                  key: azure_client_id
+            - name: AZURE_CLIENT_SECRET
+              valueFrom:
+                secretKeyRef:
+                  name: azure-cloud-credentials
+                  key: azure_client_secret
+                  optional: true
+            - name: AZURE_TENANT_ID
+              valueFrom:
+                secretKeyRef:
+                  name: azure-cloud-credentials
+                  key: azure_tenant_id
+                  optional: true
+            - name: AZURE_FEDERATED_TOKEN_FILE
+              valueFrom:
+                secretKeyRef:
+                  name: azure-cloud-credentials
+                  key: azure_federated_token_file
+                  optional: true
+          volumeMounts:
+            - name: host-etc-kube
```

Use directly from the configmap rather than the cloud.conf on the host
```suggestion
            - name: config-accm
```